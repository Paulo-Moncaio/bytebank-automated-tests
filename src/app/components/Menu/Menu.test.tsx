import { render, screen } from '@testing-library/react'
import { Menu } from '..'

test('Deve renderizar um link para a pagina inicial', () => {
  render(<Menu />)
  const link = screen.getByText('Inicial')
  expect(link).toBeDefined()
})

test('Deve renderizar uma lista de 4 links', () => {
  render(<Menu />)
  const links = screen.getAllByRole('link')
  expect(links).toHaveLength(4)
})

test('Deve renderizar uma lista de 4 links', () => {
  render(<Menu />)
  const links = screen.getAllByRole('link')
  expect(links).toHaveLength(4)
})

test('Deve renderizar uma lista de links com a classe link', () => {
  render(<Menu />)
  const links = screen.getAllByRole('link')
  links.forEach((link) => {
    expect(link.className).toBe(
      'cursor-pointer text-[#111111] no-underline hover:font-semibold hover:text-black',
    )
  })
  expect(links).toMatchSnapshot()
})
