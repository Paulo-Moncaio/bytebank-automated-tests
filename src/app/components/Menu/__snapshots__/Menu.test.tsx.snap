// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar uma lista de links com a classe link 1`] = `
[
  <a
    class="cursor-pointer text-[#111111] no-underline hover:font-semibold hover:text-black"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="cursor-pointer text-[#111111] no-underline hover:font-semibold hover:text-black"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="cursor-pointer text-[#111111] no-underline hover:font-semibold hover:text-black"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="cursor-pointer text-[#111111] no-underline hover:font-semibold hover:text-black"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
